app/connector/soap-cxf/src/main/java/io/syndesis/connector/soap/cxf/auth/AuthenticationCustomizer.java

```diff
@@ -0,0 +1,138 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.connector.soap.cxf.auth;
+
+import java.util.HashMap;
+import java.util.Map;
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.CallbackHandler;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.component.cxf.ChainedCxfEndpointConfigurer;
+import org.apache.cxf.common.util.PropertyUtils;
+import org.apache.cxf.endpoint.Client;
+import org.apache.cxf.ws.security.wss4j.WSS4JOutInterceptor;
+import org.apache.wss4j.common.ConfigurationConstants;
+import org.apache.wss4j.common.ext.WSPasswordCallback;
+
+import io.syndesis.connector.soap.cxf.ComponentProperties;
+import io.syndesis.connector.soap.cxf.SoapCxfProxyComponent;
+import io.syndesis.integration.component.proxy.ComponentProxyComponent;
+import io.syndesis.integration.component.proxy.ComponentProxyCustomizer;
+
+public final class AuthenticationCustomizer implements ComponentProxyCustomizer {
+
+    @Override
+    public void customize(final ComponentProxyComponent component, final Map<String, Object> options) {
+        consumeOption(options, ComponentProperties.AUTHENTICATION_TYPE, authType -> {
+            switch (AuthenticationType.fromValue((String) authType)) {
+            case NONE:
+                // remove unused properties
+                options.remove(ComponentProperties.USERNAME);
+                options.remove(ComponentProperties.PASSWORD);
+                options.remove(ComponentProperties.ADD_TIMESTAMP);
+                options.remove(ConfigurationConstants.ADD_USERNAMETOKEN_NONCE);
+                options.remove(ConfigurationConstants.ADD_USERNAMETOKEN_CREATED);
+                break;
+            case BASIC:
+                // assume username and password are set, and pass through
+                options.remove(ComponentProperties.ADD_TIMESTAMP);
+                options.remove(ConfigurationConstants.ADD_USERNAMETOKEN_NONCE);
+                options.remove(ConfigurationConstants.ADD_USERNAMETOKEN_CREATED);
+                break;
+            case WSSE_UT:
+                addUsernameTokenConfigurer(component, options);
+                break;
+            }
+        });
+    }
+
+    private static void addUsernameTokenConfigurer(ComponentProxyComponent component, Map<String, Object> options) {
+
+        final CamelContext camelContext = component.getCamelContext();
+        final String username = getResolvedValue(camelContext, (String) options.remove(ComponentProperties.USERNAME));
```

Comment:
Yeah, I wasn't sure about username, but I just wanted it to be uniform. The password is actually a reference, which I discovered in the unit test. The way the proxy component works, secrets are passed along as a referenced property. Which works in other components, because the default endpoint code resolved placeholders. 
But since the property is being accessed in the customizer, I have to resolve it there just like the default endpoint.

Reply:
That doesn't seem right, we do it differently [here](https://github.com/syndesisio/syndesis/blob/master/app/connector/rest-swagger/src/test/java/io/syndesis/connector/rest/swagger/RestSwaggerConnectorIntegrationTest.java)

Reply:
The difference might be that the value is later resolved by underlying Camel bits. Here the value (which is not resolved yet in Camel) is passed directly to a CXF interceptor, which later fails.